### PR TITLE
Idefix Menu "Preview Display Mode"

### LINKED ISSUE #2760: provide option to change preview display mode from idefix > preview menu entry
### Describe the feature and the current behavior/state
currently, under idefix > preview, it shows the preview in the display mode defined in the texstudio configuration .

the proposal is to have this ability to change the preview display mode on the fly from menu items too

### Who will benefit with this feature?

anyone using live instant preview for different use cases (like normal latex, or math equations, or tikzfigure, etc)

### Any Other info

* TeXstudio 4.3.1 (git 4.3.1)
* Using Qt Version 6.3.1, compiled with Qt 6.3.1 R
* Windows 10 20H2

### PR DESCRIPTION
**Pull request by octaeder**
This PR adds a preview options group menu to the idefix menu and resolves #2760.

grafik

### CODE REVIEW

**Comment by octaeder** [comment]
@sunderme  Can you help with the broken CI? What should line 1473

configManager.previewMode = act->data().value<ConfigManager::PreviewMode>();

look like?

File: src/texstudio.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@
  */
 
 
-
+QActionGroup *previewModeGroup = nullptr;
```
Comment (sunderme) [inline src/texstudio.cpp:113]: there is no need to use a global variable here.
global variables are considered bad in cpp.

File: src/texstudio.cpp [hunk 2]
```diff
@@ -1057,9 +1057,45 @@ void Texstudio::setupMenus()
...
     newManagedAction(menu, "previewLatex", tr("Pre&view Selection/Parentheses"), SLOT(previewLatex()), Qt::ALT | Qt::Key_P);
 	newManagedAction(menu, "removePreviewLatex", tr("C&lear Inline Preview"), SLOT(clearPreview()));
 
+	submenu = newManagedMenu(menu, "previewMode", tr("Preview Dis&play Mode"));
+	previewModeGroup = new QActionGroup(this);
+	act = newManagedAction(submenu, "preview1", tr("Show preview as tooltip if panel is hidden"), SLOT(setPreviewMode()));
+	act->setData(ConfigManager::PM_TOOLTIP_AS_FALLBACK);
+	act->setCheckable(true);
+	previewModeGroup->addAction(act);
+	act = newManagedAction(submenu, "preview2", tr("Always show preview in preview panel"), SLOT(setPreviewMode()));
```
Comment (sunderme) [inline src/texstudio.cpp:1071]: menu names should be, at least a little, self-explanatory. 
preview2 (etc.) is definitely not. 
There is a reason why the enum is called PM_PANEL.

File: src/texstudio.cpp [hunk 3]
```diff
@@ -1057,9 +1057,45 @@ void Texstudio::setupMenus()
...
+	previewModeGroup->addAction(act);
+	act = newManagedAction(submenu, "preview4", tr("Always show both"), SLOT(setPreviewMode()));
+	act->setData(ConfigManager::PM_BOTH);
+	act->setCheckable(true);
+	previewModeGroup->addAction(act);
+	act = newManagedAction(submenu, "preview5", tr("Inline"), SLOT(setPreviewMode()));
+	act->setData(ConfigManager::PM_INLINE);
+	act->setCheckable(true);
+	previewModeGroup->addAction(act);
+	act = newManagedAction(submenu, "preview6", tr("Show in embedded viewer"), SLOT(setPreviewMode()));
```
Comment (sunderme) [inline src/texstudio.cpp:1087]: if you look at `configdialog.cpp` ll. 584..590, the last option is not present if no pdf previewer is build.

**Comment by sunderme** [comment]
in configmanager.h `Q_DECLARE_METATYPE(ConfigManager::PreviewMode);` is missing at one before last line. (CI failure)

**Comment by sunderme** [comment]
and for consistency, I would actually prefer to stick to txs way of using menus.
An example which basically does the exact same thing is "line ending" (menu edit).
see texstudio.cpp lines 1849-1867  and 1028ff

**Comment by octaeder** [comment]
hope this is the last change. Thanks for your support.

**Comment by sunderme** [comment]
thanks

**Comment by octaeder** [comment]
@sunderme I see that the translation misses the accelarator key (&p) for the submenu (s. image above). Is this intended?

grafik

**Comment by sunderme** [comment]
no

**Pull request merged**